BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
+}
+
+public struct Rate
+{
+    [JsonPropertyName("currencyCode")] public string CurrencyCode { get; set; }
+    [JsonPropertyName("rate")] public float RateValue { get; set; }
+}
+
+internal class ExchangeRateService
+{
+    private const string MainCurrenciesUrl = "https://api.cnb.cz/cnbapi/exrates/daily";
+
+    public IEnumerable<Rate> GetExchangeRates(List<string> currencies)
```

Comment:
* This list of currencies should come from the request body, assuming the client will always inform the currencies that they want to know the rates.
* I would create a class to be placed in the application layer to represent the input coming from the request. A suggestion could be:

```
public struct GetExchangeRatesRequest
{
    [JsonPropertyName("currencies")]
    public IEnumerable<string> TargetCurrenciesList { get; set; }
}
```

Reply:
* This list of currency codes should come from the request body, assuming the client will always inform the currencies that they want to know the rates. 
* I would create a class to be placed in the application layer to represent the input coming from the request. A suggestion could be:

```
public struct GetExchangeRatesRequest
{
    [JsonPropertyName("currencies")]
    public IEnumerable<string> TargetCurrencyCodeList { get; set; }
}
```

Reply:
Still regarding the input from the request body, we should validate it to avoid errors. In this case, we could consider two options:

1. Creating a `List<string> TargetCurrencyCodes` property inside the **CnbOptions.cs** (we could create a different option class for this, but I would avoid increasing complexity at the moment). This would allow us to verify, for example, within the service, if the currency codes informed by the client are in the list we specified in **appsettings.json**
2. Using **Fluent Validation**: the input is validated as soon as the request enters the controller, but before the logic is executed. Docs: https://docs.fluentvalidation.net/en/latest/start.html

In both cases, we can configure an error to be shown to the client.